bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/api/BondHttpApi.java

```diff
@@ -0,0 +1,246 @@
+     * @return the {@link BondDeviceProperties}
+     * @throws IOException
+     */
+    @Nullable
+    public BondDeviceProperties getDeviceProperties(String deviceId) throws IOException {
+        String json = request("/v2/devices/" + deviceId + "/properties");
+        logger.trace("BondHome device properties : {}", json);
+        return gson.fromJson(json, BondDeviceProperties.class);
+    }
+
+    /**
+     * Executes a device action
+     *
+     * @param deviceId The ID of the device
+     * @param actionId The Bond action
+     * @param argument An additional argument for the actions (such as the fan speed)
+     */
+    public synchronized void executeDeviceAction(String deviceId, BondDeviceAction action, @Nullable Integer argument) {
+        String url = "http://" + bridgeHandler.getBridgeIpAddress() + "/v2/devices/" + deviceId + "/actions/"
+                + action.getActionId();
+        String payload = "{}";
+        if (argument != null) {
+            payload = "{\"argument\":" + argument + "}";
+        }
+        InputStream content = new ByteArrayInputStream(payload.getBytes());
+        try {
+            logger.debug("HTTP PUT to {} with content {}", url, payload);
+
+            Properties headers = new Properties();
+            headers.put("BOND-Token", bridgeHandler.getBridgeToken());
+
+            String httpResponse = HttpUtil.executeUrl(HttpMethod.PUT, url, headers, content, "application/json",
+                    BOND_API_TIMEOUT_MS);
+            logger.debug("HTTP response from {}: {}", deviceId, httpResponse);
+        } catch (IOException ignored) {
+            logger.warn("Unable to execute device action!");
+        }
+    }
+
+    /**
+     * Submit GET request and return response, check for invalid responses
+     *
+     * @param uri: URI (e.g. "/settings")
+     */
+    private synchronized String request(String uri) throws IOException {
+        String httpResponse = "ERROR";
+        String url = "http://" + bridgeHandler.getBridgeIpAddress() + uri;
+        int numRetriesRemaining = 3;
+        do {
+            try {
+                logger.debug("HTTP GET to {}", url);
+
+                final Properties headers = new Properties();
+                headers.put("BOND-Token", bridgeHandler.getBridgeToken());
+
+                httpResponse = HttpUtil.executeUrl(HttpMethod.GET, url, headers, null, "", BOND_API_TIMEOUT_MS);
+                if (httpResponse == null) {
+                    throw new IOException("No response received!");
+                }
+                // handle known errors
+                if (httpResponse.contains(API_ERR_HTTP_401_UNAUTHORIZED)) {
+                    // Don't retry or throw an exception if we get unauthorized, just set the bridge offline
+                    numRetriesRemaining = 0;
+                    bridgeHandler.setBridgeOffline(ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Incorrect local token for Bond Bridge.");
```

Comment:
Sorry for vague advise. This can be simplified to:
```java
bridgeHandler.setBridgeOffline(ThingStatusDetail.CONFIGURATION_ERROR,
                            "@text/offline.conf-error.incorrect-local-token");
```

Then in bondhome.properties:

```
# thing status descriptions

offline.conf-error.incorrect-local-token = Incorrect local token for Bond Bridge.
```

Reply:
heh, well, that's certainly a lot easier than plumbing my own translation provider through :).

Reply:
Indeed. :)